src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -844,7 +844,7 @@ private string ReadName(int position, int nameLength)
                 if (checkErrors)
                 {
                     long result;
-                    while (Interop.CheckIo(result = sysCall(_inotifyHandle, arg1, arg2), isDirectory: true)) ;
+                    Interop.CheckIo(result = sysCall(_inotifyHandle, arg1, arg2), isDirectory: true);
```

Comment:
I've removed it.  It turns out there was only 1 call to SysCall that was actually doing any checking.